src/libs/ajax.js

```diff
@@ -1153,20 +1153,21 @@ const Buckets = signal => ({
     return res.json()
   },
 
-  listAll: async (googleProject, bucket, prefix = null, pageToken = null) => {
+  listAll: async (googleProject, bucket, prefix = null, pageToken = null, delimiter = null) => {
     const res = await fetchBuckets(
-      `storage/v1/b/${bucket}/o?${qs.stringify({ prefix, pageToken })}`,
+      `storage/v1/b/${bucket}/o?${qs.stringify({ prefix, delimiter, pageToken })}`,
       _.merge(authOpts(await saToken(googleProject)), { signal })
     )
     const body = await res.json()
     const items = body.items || []
+    const prefixes = body.prefixes || []
```

Comment:
A few different parts of Terra UI use `listAll`. None of the existing usages care about the `prefixes` that can be collected while iterating through the pages, but because the workspace file browser displays files in "directory mode" we want to collect the prefixes so we can display them as folders and allow users to click into them